Sources/MongoSwift/ClientSession.swift

```diff
@@ -1,35 +1,106 @@
+import Foundation
+import mongoc
+
+internal typealias MutableClientSessionPointer = OpaquePointer
+
 /// Options to use when creating a ClientSession.
-internal struct SessionOptions: Encodable {
-    /// Specifies whether read operations should be causally ordered within the session.
-    private let causalConsistency: Bool?
+public struct ClientSessionOptions {}
+
+/// Private helper for providing a `mongoc_session_opt_t` that is only valid within the body of the provided
+/// closure.
+private func withSessionOpts<T>(wrapping options: ClientSessionOptions?,
+                                _ body: (OpaquePointer) throws -> T) rethrows -> T {
```

Comment:
I'd actually argued against these before. see [here](https://github.com/mongodb/swift-mongo-mobile/pull/14#discussion_r268257188).

Reply:
as an addition to that comment, I would say that it was useful to typealias in the cases where a pointer's type is different across Swift versions, for the bson_ts and iterators. I'm just a bit skeptical about ending up with like 30 typealiases that all are ultimately just `OpaquePointer`s.

Reply:
I think my primary motivation is readability here. We don't actually _lose_ anything from type aliasing, and I think we gain an ability to keep our Swift code in line with what's going on in mongoc. If the goal of keeping it an `OpaquePointer` is to warn the developer that they need to go open the mongoc docs and triple check everything when dealing with it, then we could name it `UnsafeMongocSessionOptsPointer` or something - even that gives me more context than `OpaquePointer`. Are there other benefits to keeping it `OpaquePointer`?

Reply:
I agree that it increases readability to know what the pointer is supposedly pointing to. I just feel weird about the fact that any unimported libmongoc type can be implicitly cast to a `SessionOptsPointer`.
I think it's easy to be like "oh, this function takes in a real type, and it compiles when I pass in this variable, so my variable must be the right type" even though you passed in a pointer to a bson_t or something. it seems like if you know it's opaque you might be more likely to look into what exactly it's supposed to be pointing to.... maybe that's just how I think about these though.

maybe we could prefix the typealiases with `Opaque`? `OpaqueSessionOptsPointer` etc.

Reply:
another thought I had on these is that in cases we only use them one time, there is probably another way to get the same kind of readability without introducing an extra symbol.
for example, in this case we can just add a parameter label in the closure, like 
```swift
private func withSessionOpts<T>(wrapping options: ClientSessionOptions?,
                                _ body: (_ sessionOptsPtr: OpaquePointer) throws -> T) rethrows -> T {
// ...
}
```

Reply:
another thing to consider is actually just naming stored opaque pointer properties in a way that describes what they are -- for ex., `ClientSession` below has 
```swift
 internal fileprivate(set) var _session: MutableClientSessionPointer?
```

if we renamed `_session` -> `mutableClientSessionPointer` then you would get the readability benefit everyplace you use it (for ex. the `init`, in `advanceClusterTime`), not just at the place where you define the property.

then again we generally don't include the type name in other stored properties, so, idk. just a thought.

Reply:
Let's just leave it as it. If the pointer is being used in difficult situations (like the ones needed for Swift 5 support) then we'll type alias them, otherwise use `OpaquePointer`

Reply:
In line with this, I've removed the `ClientSessionMutablePointer` typealias.